core/application/reconcile.go

```diff
@@ -52,52 +57,99 @@ func (a *app) ResyncMicroVMs(ctx context.Context, namespace string) error {
 	return nil
 }
 
-func (a *app) plan(spec *models.MicroVM, logger *logrus.Entry) (planner.Plan, error) {
+func (a *app) plan(spec *models.MicroVM, logger *logrus.Entry) planner.Plan {
 	l := logger.WithField("stage", "plan")
 	l.Info("Generate plan")
 
-	if spec.Status.Retry > a.cfg.MaximumRetry {
-		return nil, reachedMaximumRetryError{vmid: spec.ID, retries: spec.Status.Retry}
-	}
-
 	// Delete only if the spec was marked as deleted.
 	if spec.Spec.DeletedAt != 0 {
 		input := &plans.DeletePlanInput{
 			StateDirectory: a.cfg.RootStateDir,
 			VM:             spec,
 		}
 
-		return plans.MicroVMDeletePlan(input), nil
+		return plans.MicroVMDeletePlan(input)
 	}
 
 	input := &plans.CreateOrUpdatePlanInput{
 		StateDirectory: a.cfg.RootStateDir,
 		VM:             spec,
 	}
 
-	return plans.MicroVMCreateOrUpdatePlan(input), nil
+	return plans.MicroVMCreateOrUpdatePlan(input)
+}
+
+func (a *app) reschedule(ctx context.Context, logger *logrus.Entry, spec *models.MicroVM) error {
+	spec.Status.Retry++
+	waitTime := time.Duration(spec.Status.Retry*backoffBaseInSeconds) * time.Second
+	spec.Status.NotBefore = time.Now().Add(waitTime).Unix()
+
+	logger.Infof(
+		"[%d/%d] reconciliation failed, reschedule at %s",
+		spec.Status.Retry,
+		a.cfg.MaximumRetry,
+		time.Unix(spec.Status.NotBefore, 0),
+	)
+
+	if _, err := a.ports.Repo.Save(ctx, spec); err != nil {
+		return fmt.Errorf("saving spec after plan failed: %w", err)
+	}
+
+	go func(id, ns string, sleepTime time.Duration) {
+		logger.Info("Wait to emit update")
+		time.Sleep(sleepTime)
+		logger.Info("Emit update")
```

Comment:
do we want to be a bit more clear on these logs?

like "waiting to publish update event" or "waiting to reschedule for update"

Reply:
I don't think we even need these lines, added them while i was working on it.